src/Lower.cpp

```diff
@@ -80,6 +80,18 @@ using std::ostringstream;
 using std::string;
 using std::vector;
 
+namespace {
+
+Stmt new_or_empty(const Stmt &s, Stmt &last_written) {
+    if (s.same_as(last_written)) {
+        return Stmt();
```

Comment:
Would be nice if we could emit something like `(Unchanged)` for this case, rather than an empty Stmt, but that's just extra-credit.